course_info/static/course_info/js/controllers/DetailsController.js

```diff
@@ -38,22 +38,27 @@
 
         dc.associateNewSite = function() {
             dc.associateNewSiteInProgress = true;
-            // todo: call backend
-            $timeout(function() {
-                var fakeCourse = {
+            data = {
+              'external_id': dc.newCourseSiteURL
+            };
+            var post_course_site_url = djangoUrl.reverse(dc.apiProxy,
+                        [dc.apiBase + dc.courseInstance.course_instance_id + '/sites/']);
+            $http.post(post_course_site_url, data).then(function(response){
+                var new_course = {
                     course_site_url: dc.newCourseSiteURL,
-                    map_type: 'official'
+                    map_type: 'official',
+                    site_map_id: response.data.site_map_id
                 };
-                dc.courseInstance.sites.push(fakeCourse);
+                dc.courseInstance.sites.push(new_course);
                 dc.newCourseSiteURL = '';
                 dc.associateNewSiteInProgress = false;
```

Comment:
I don't think I understand what you mean. I think we only want to push a new site onto the list if the post was successful? no?